### PR TITLE
Add capture to open file descriptors in system metrics instrumentation

### LINKED ISSUE #2562: looks like process.open_file_descriptor.count is not implemented in python
**Is your feature request related to a problem?**

I do not see `process.open_file_descriptor.count` metric being collected in https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py. 

[This](https://opentelemetry.io/docs/specs/semconv/system/process-metrics/#metric-processopen_file_descriptorcount) says that it should be collected.

**Describe the solution you'd like**

`process.open_file_descriptor.count` metric should be collected.

**Describe alternatives you've considered**

N/A
**Additional context**

N/A

### PR DESCRIPTION
**Pull request by emdneto**
# Description

Fixes #2562

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] tox 

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [X] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [X] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [X] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by xrmx — COMMENTED** [review]
This requires a changelog entry

**Review by xrmx — APPROVED** [review]

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 1]
```diff
@@ -395,9 +397,25 @@ def _instrument(self, **kwargs):
                 unit="switches",
             )
 
+        if "process.open_file_descriptor.count" in self._config:
+            self._meter.create_observable_up_down_counter(
+                name=f"process.{self._python_implementation}.open_file_descriptor.count",
+                callbacks=[self._get_open_file_descriptors],
+                description="Number of open file descriptors",
```
Comment (lzchen) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:404]:
```suggestion
                description="Number of file descriptors in use by the process.",
```

File: instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py [hunk 2]
```diff
@@ -395,9 +397,25 @@ def _instrument(self, **kwargs):
                 unit="switches",
             )
 
+        if "process.open_file_descriptor.count" in self._config:
+            self._meter.create_observable_up_down_counter(
+                name=f"process.{self._python_implementation}.open_file_descriptor.count",
```
Comment (xrmx) [inline instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:402]: Taking a second look I don't think this is dependent on the python implementation though. Do we need it? Genuine question, no clue why we are adding these prefixes.

Comment (emdneto) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:402]: Good question. If we look at [semconv here](https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-semantic-conventions/src/opentelemetry/semconv/_incubating/metrics/process_metrics.py#L177) there's no reference to python implementation in the metric name.

Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:402]: It does look like the original intent was to match the semantic conventions for [runtime environment metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/README.md#metrics) where the runtime is prefixed. I believe the question is whether `process.*` metrics should be treated as runtime metrics or `process` metrics. To me it is mostly the latter so we should probably follow the semantic conventions for [process metrics](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/system/process-metrics.md#process-metrics) and remove the runtime prefix altogether.

Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py:402]: @emdneto 

For this pr, I suggest removing the prefix for file descriptor and maybe opening up an issue to remove the prefixes for everything else is people agree on this approach.

Comment (emdneto) [reply instrumentation/opentelemetry-instrumentation-system-metrics/src/opentelemetry/instrumentation/system_metrics/__init__.py]: I'll open the issue. Thanks

**Review by ocelotl — APPROVED** [review]

**Pull request merged**